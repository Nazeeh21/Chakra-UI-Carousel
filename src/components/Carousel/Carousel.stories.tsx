import { Box, Text } from "@chakra-ui/react";
import { Story } from "@storybook/react";
import React from "react";
import Carousel from ".";

export default {
  title: "Carousel",
  component: Carousel,
  parameters: {
    chromatic: {
      disableSnapshot: false,
    },
  },
};

export const DemoComp = ({ index }: { index: number }) => {
  return (
    <Box borderWidth={2} borderColor="black" w={"10rem"}>
      <Text>Component {index}</Text>
    </Box>
  );
};

export const Primary: Story = () => (
  <Carousel gap={3}>
    <Box>Carousel</Box> <Box>Carousel 2</Box>
  </Carousel>
);
