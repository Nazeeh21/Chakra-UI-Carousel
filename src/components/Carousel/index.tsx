import { Flex } from "@chakra-ui/react";
import React from "react";

export interface CarouselPropTypes {
  children: React.ReactNode[];
  gap: number;
}

const Carousel: React.FC<CarouselPropTypes> = ({ children, gap }) => {
  return <Flex gap={gap}>{children}</Flex>;
};

export default Carousel;
