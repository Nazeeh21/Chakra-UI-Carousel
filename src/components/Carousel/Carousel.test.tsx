import React from "react";
import Carousel from ".";
import renderer from "react-test-renderer";
import { DemoComp } from "./Carousel.stories";

describe("renders correctly", () => {
  it("Carousel renders correctly", () => {
    const container = renderer
      .create(
        <Carousel gap={3}>
          <DemoComp index={1} />
          <DemoComp index={2} />
          <DemoComp index={3} />
          <DemoComp index={4} />
        </Carousel>
      )
      .toJSON();
    expect(container).toMatchSnapshot();
  });
});
