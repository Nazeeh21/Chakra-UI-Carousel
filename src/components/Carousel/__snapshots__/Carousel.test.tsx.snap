// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Carousel renders correctly 1`] = `
<div
  className="css-blvggq"
>
  <div
    className="css-18oaxk3"
  >
    <p
      className="chakra-text css-0"
    >
      Component 
      1
    </p>
  </div>
  <div
    className="css-18oaxk3"
  >
    <p
      className="chakra-text css-0"
    >
      Component 
      2
    </p>
  </div>
  <div
    className="css-18oaxk3"
  >
    <p
      className="chakra-text css-0"
    >
      Component 
      3
    </p>
  </div>
  <div
    className="css-18oaxk3"
  >
    <p
      className="chakra-text css-0"
    >
      Component 
      4
    </p>
  </div>
</div>
`;
